xroad-catalog-lister/src/main/java/org/niis/xroad/catalog/lister/configuration/ConfigClientInitializer.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * The MIT License
+ *
+ * Copyright (c) 2023- Nordic Institute for Interoperability Solutions (NIIS)
+ * Copyright (c) 2016-2023 Finnish Digital Agency
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.catalog.lister.configuration;
+
+import org.niis.xroad.confclient.config.ConfClientRootConfig;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.annotation.Import;
+import org.springframework.stereotype.Component;
+
+@Import(ConfClientRootConfig.class)
+@Component
+@DependsOn("configClientPropertiesLoader")
+public class ConfigClientInitializer {
```

Comment:
It is probably not a good practice to have a dummy component with @import of additional configurations. Generally to achieve what you want we would use EnvironmentPostProcessor. We have examples in xroad (refer to DatabasePropertiesEnvironmentPostProcessor), take note that it has to be registered in spring factories.

Reply:
@ricardas-buc I modified this class by:
* fixing it's type (modified it from component to configuration)
* adding a new feature flag that is set by default in the application.yaml to enable configuration-client and disabled it in tests
* removed the dummy version from tests